src/util-ebpf.c

```diff
@@ -388,61 +559,87 @@ static int EBPFBypassedFlowV6Timeout(int fd, struct flowv6_keys *key,
  */
 static int EBPFForEachFlowV4Table(LiveDevice *dev, const char *name,
                                   struct flows_stats *flowstats,
-                                  struct timespec *ctime)
+                                  struct timespec *ctime,
+                                  struct ebpf_timeout_config *tcfg,
+                                  OpFlowForKey EBPFOpFlowForKey
+                                  )
 {
     int mapfd = EBPFGetMapFDByName(dev->dev, name);
     struct flowv4_keys key = {}, next_key;
     int found = 0;
     unsigned int i;
-    unsigned int nr_cpus = UtilCpuGetNumProcessorsConfigured();
-    if (nr_cpus == 0) {
-        SCLogWarning(SC_ERR_INVALID_VALUE, "Unable to get CPU count");
+    uint64_t hash_cnt = 0;
+
+    if (tcfg->cpus_count == 0) {
+        SCLogWarning(SC_ERR_INVALID_VALUE, "CPU count should not be 0");
         return 0;
     }
 
-    uint64_t hash_cnt = 0;
+    uint64_t pkts_cnt = 0;
     while (bpf_map_get_next_key(mapfd, &key, &next_key) == 0) {
-        bool purge = true;
-        uint64_t pkts_cnt = 0;
         uint64_t bytes_cnt = 0;
         hash_cnt++;
-        /* We use a per CPU structure so we will get a array of values. */
-        struct pair values_array[nr_cpus];
+        if (pkts_cnt > 0) {
+            SC_ATOMIC_ADD(dev->bypassed, pkts_cnt);
+            EBPFDeleteKey(mapfd, &key);
+        }
+        pkts_cnt = 0;
+        /* We use a per CPU structure so we will get a array of values. But if nr_cpus
+         * is 1 then we have a global hash. */
+        BPF_DECLARE_PERCPU(struct pair, values_array, tcfg->cpus_count);
         memset(values_array, 0, sizeof(values_array));
```

Comment:
does this take into account the PERCPU in its size?

Reply:
The macro adds alignment instruction in the structure declaration so it should be seen when using sizeof().